tests/tests.cmake

```diff
@@ -207,29 +207,35 @@ function(add_generic_test)
         add_custom_target(${arch}-${test_name}-phys-yaml DEPENDS ${phys_yaml})
 
         # Output FASM target
-        set(fasm ${output_dir}/${name}.fasm)
-        add_custom_command(
-            OUTPUT ${fasm}
-            COMMAND ${CMAKE_COMMAND} -E env
-                ${quiet_cmd}
-                ${PYTHON3} -mfpga_interchange.fasm_generator
-                    --schema_dir ${INTERCHANGE_SCHEMA_PATH}
-                    --family ${arch}
-                    ${device_loc}
+        if(${arch} STREQUAL "xcup")
+            # FASM not supported
+            add_dependencies(all-tests ${arch}-${test_name}-phys)
+            add_dependencies(all-${device}-tests ${arch}-${test_name}-phys)
+        else()
+            set(fasm ${output_dir}/${name}.fasm)
+            add_custom_command(
+                OUTPUT ${fasm}
+                COMMAND ${CMAKE_COMMAND} -E env
+                    ${quiet_cmd}
+                    ${PYTHON3} -mfpga_interchange.fasm_generator
+                        --schema_dir ${INTERCHANGE_SCHEMA_PATH}
+                        --family ${arch}
+                        ${device_loc}
+                        ${netlist}
+                        ${phys}
+                        ${fasm}
+                DEPENDS
+                    ${device_target}
+                    ${arch}-${test_name}-netlist
+                    ${arch}-${test_name}-phys
                     ${netlist}
                     ${phys}
-                    ${fasm}
-            DEPENDS
-                ${device_target}
-                ${arch}-${test_name}-netlist
-                ${arch}-${test_name}-phys
-                ${netlist}
-                ${phys}
-        )
+            )
 
-        add_custom_target(${arch}-${test_name}-fasm DEPENDS ${fasm})
-        add_dependencies(all-tests ${arch}-${test_name}-fasm)
-        add_dependencies(all-${device}-tests ${arch}-${test_name}-fasm)
+            add_custom_target(${arch}-${test_name}-fasm DEPENDS ${fasm})
+            add_dependencies(all-tests ${arch}-${test_name}-fasm)
+            add_dependencies(all-${device}-tests ${arch}-${test_name}-fasm)
+        endif()
```

Comment:
I think it might be worth emitting the command either way and just handle the dependency addition to `all-tests` with the if-else, so that, when developing the FASM generator, the cmake structure can be left untouched upon running the fasm command.

Reply:
Thanks, fixed